### PR TITLE
Update a wishlist model

### PR DESCRIPTION
**Pull request by Dracco30**

### CODE REVIEW

File: service/models.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def update(self):
         """
         Updates a Wishlist to the database
         """
-        logger.info("Saving %s", self.id)
+        logger.info("Updating %s", self.name)
```
Comment (studentgrad0419) [inline service/models.py:60]: Hi, this causes many tests to fail bcause items do not have a "name attribute"

File: tests/test_routes.py [hunk 2]
```diff
@@ -190,6 +190,22 @@ def test_get_wishlist_list(self):
...
+
+        # update the wishlist
+        new_wishlist = resp.get_json()
+        new_wishlist["name"] = "devops-wishlist"
+        new_wishlist_id = new_wishlist["id"]
+        resp = self.client.put(f"{BASE_URL}/{new_wishlist_id}", json=new_wishlist)
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        updated_wishlist = resp.get_json()
+        self.assertEqual(updated_wishlist["name"], "devops-wishlist")
+
```
Comment (studentgrad0419) [inline tests/test_routes.py:208]: the wishlist has an attribute "wishlist_name" not simply "name"

**Review by studentgrad0419 — CHANGES_REQUESTED** [review]
Have you tested your code with green? I have run it and there's many failing tests.
I think I found the key areas and put comments of what can fix these issues.

**Comment by Dracco30** [comment]
> Have you tested your code with green? I have run it and there's many failing tests. I think I found the key areas and put comments of what can fix these.

I have tested it, why it says that many have no attribute name? Right now I have no idea how to modify this.

**Comment by Dracco30** [comment]
> I have fixed everything by changing "self.name" -> "self.id" in models.py and "name" to -> "wishlist_name" in test_models.py that is all you need to do to fix this.

I have made the change, the test accuracy is 97% now.

**Comment by studentgrad0419** [comment]
Okay, there's one thing I missed last time and it's the sad path (you want to test the 404 not found path for update a wishlist)

**Comment by Dracco30** [comment]
> Okay, there's one thing I missed last time and it's the test for the sad path (you want to test the 404 not found path for update a wishlist)

Sorry, but I do not understand this, what change need to be made?

**Comment by studentgrad0419** [comment]
> > Okay, there's one thing I missed last time and it's the test for the sad path (you want to test the 404 not found path for update a wishlist)
> 
> Sorry, but I do not understand this, what change need to be made?

One more test method needs to be made (updating a wishlist that does not exist) should return a 404 not found status.

File: tests/test_routes.py
Comment (studentgrad0419) [inline tests/test_routes.py:1]: This should pass for the sprint, might consider an alternative to a hard coded id to check

**Review by studentgrad0419 — APPROVED** [review]
Looks good to merge.
Thanks for the contributions!

**Pull request merged**